File: sorting_algos/Bubble_sort.py
# Python program for implementation of Bubble Sort

def bubbleSort(array):
	n = len(array)

	# Traverse through all array elements
	for i in range(n-1):
	# range(n) also work but outer loop will repeat one time more than needed.

		# Last i elements are already in place
		for j in range(0, n-i-1):

			# traverse the array from 0 to n-i-1
			# Swap if the element found is greater
			# than the next element
			if array[j] > array[j+1] :
				array[j], array[j+1] = array[j+1], array[j]

# Driver code to test above
array = [64, 34, 25, 12, 22, 11, 90]

bubbleSort(array)

print ("Sorted array is:")
for i in range(len(array)):
	print ("%d" %array[i]),

File: sorting_algos/bubble_sort.cpp
// C++ Program for implementaion of Bubble sort
// Bubble Sort is the process of sorting adjacent elements

#include <iostream>
using namespace std;
int main()
{
    int arr[] = {12, 45, 8, 5, 16};
    int n = sizeof(arr) / sizeof(arr[0]);
    int temp, i, j;

    // Displaying original array
    cout << "\n The array is : \n";
    for (i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }

    // Bubble sort
    for (i = 0; i < n; i++)
    {
        for (j = 0; j < n - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }

    // Displaying sorted array
    cout << "\n After sorting : \n";
    for (i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }

    return 0;
}